backend/app/api/endpoints/internal/workspace_archives.py

```
@@ -0,0 +1,87 @@
+class ManualArchiveResponse(BaseModel):
+    """Response model for manually triggered workspace archive."""
+
+    success: bool = True
+    task_id: int
+    archive: ArchiveInfo
+
+
+@router.post("/{task_id}/archive", response_model=ManualArchiveResponse)
+async def archive_task_workspace(
+    task_id: int,
+    db: Session = Depends(get_db),
```

Ruff (0.15.7)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)